PICMI_Python/diagnostics.py

```diff
@@ -2,14 +2,32 @@
 These should be the base classes for Python implementation of the PICMI standard
 The classes in the file are all diagnostics related
 """
+import typing
+from collections.abc import Sequence
+
+from autoclass import autoargs
+from typeguard import typechecked
 
 from .base import _ClassWithInit
+from . import fields
+from . import particles
+
+GridType = typing.NewType('GridType', typing.Union[fields.PICMI_Cartesian1DGrid,
+                                                   fields.PICMI_CylindricalGrid,
+                                                   fields.PICMI_Cartesian2DGrid,
+                                                   fields.PICMI_Cartesian3DGrid])
+
+SpeciesType = typing.NewType('SpeciesType', typing.Union[particles.PICMI_Species,
+                                                         particles.PICMI_MultiSpecies])
+SpeciesArgument = typing.NewType('SpeciesArgument', typing.Union[SpeciesType, Sequence[SpeciesType]])
```

Comment:
same as above: How do you feel about collecting all type definitions in a single file?

I am worried that existing type definitions might be overlooked and defined over and over again.